### PR TITLE
feat: decouple event constraints from deserialization

### LINKED ISSUE #15208: Decouple event constraints from deserialization

### PR DESCRIPTION
**Pull request by lpetrovic05**
closes #15208

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java [hunk 1]
```diff
@@ -157,16 +171,57 @@ public DefaultInternalEventValidator(
...
 
         return true;
     }
 
+    /**
+     * Checks whether the transaction is null.
+     * @param transaction the transaction to check
+     * @return true if the transaction is null, otherwise false
+     */
+    private static boolean isTransactionNull(@Nullable final EventTransaction transaction) {
```
Comment (litt3) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:194]: method is named `isTransactionNull`, but it actually returns true if the transaction *isn't* null

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java [hunk 2]
```diff
@@ -157,16 +171,57 @@ public DefaultInternalEventValidator(
      * @return true if the required fields of the event are non-null, otherwise false
      */
     private boolean areRequiredFieldsNonNull(@NonNull final PlatformEvent event) {
```
Comment (litt3) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:173]: It wasn't already in this class, but should we also be checking `maxTransactionCountPerEvent` and `transactionMaxBytes` here? 

`maxTransactionCountPerEvent` currently isn't being validated *anywhere*, and `transactionMaxBytes` isn't checked anywhere in intake (it's only checked on the node submitting the transaction)

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java [hunk 3]
```diff
@@ -157,16 +171,57 @@ public DefaultInternalEventValidator(
...
-        if (event.getSignature() == null) {
-            // do not log the event itself, since toString would throw a NullPointerException
-            nullUnhashedDataLogger.error(EXCEPTION.getMarker(), "Event has null signature");
-            nullUnhashedDataAccumulator.update(1);
+        final GossipEvent gossipEvent = event.getGossipEvent();
+        final EventCore eventCore = gossipEvent.eventCore();
+        if (eventCore == null
```
Comment (litt3) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:176]: Each of these `null` cases should have a unit test, to make sure it causes validation to fail

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java (hunk 1 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:194]: yeah, missed that. added unit tests and fixed it

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java [hunk 4]
```diff
@@ -157,16 +171,59 @@ public DefaultInternalEventValidator(
...
+    /**
+     * Checks whether the {@link com.hedera.pbj.runtime.io.buffer.Bytes} fields of an event are the expected length.
+     *
+     * @param event the event to check
+     * @return true if the byte fields of the event are the correct length, otherwise false
+     */
+    private boolean areByteFieldsCorrectLength(@NonNull final PlatformEvent event) {
+        final GossipEvent gossipEvent = event.getGossipEvent();
+        final EventCore eventCore = event.getEventCore();
+        if (gossipEvent.signature().length() != SignatureType.RSA.signatureLength()) {
```
Comment (litt3) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:209]: Here and several other places in this class are lacking coverage. Suggest using intellij coverage tool to address

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:176]: done

Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:176]: added unit tests for each of them

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java (hunk 4 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:217]: i added unit tests for this

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java (hunk 2 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:173]: I will create a separate ticket for this

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java (hunk 4 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:217]: it now has coverage for all new code

**Review by litt3 — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java [hunk 5]
```diff
@@ -173,12 +173,20 @@ public DefaultInternalEventValidator(
...
-            nullFieldLogger.error(EXCEPTION.getMarker(), "Event has null field {}", gossipEvent);
+        String nullField = null;
+        if (eventCore == null) {
+            nullField = "eventCore";
+        } else if (eventCore.timeCreated() == null) {
+            nullField = "timeCreated";
+        } else if (eventCore.version() == null) {
+            nullField = "version";
+        } else if (eventCore.parents().stream().anyMatch(Objects::isNull)) {
+            nullField = "parent";
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:184]: code coverage indicates line 184 was missed.

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java (hunk 5 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:184]: you must be looking at the old report, i added coverage for this

**Pull request merged**